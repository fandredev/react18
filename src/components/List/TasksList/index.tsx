import classNames from 'classnames';
import { TasksListProps } from "types/tasks";
import styles from './taskslist.module.scss';

interface ListTasksProps extends TasksListProps {
  selectTask: (selectedTask: TasksListProps) => void
}


export default function TasksList({ task, time, id, selected, completed, selectTask }: ListTasksProps) {
  return (
    <li className={classNames({
      [styles.item]: true,
      [styles.listSelected]: selected,
      [styles.listCompleted]: completed
    })} onClick={() => !completed && selectTask({
      task, time, id, selected, completed
    })}>
      <h3>{task}</h3>
      <span>{time}</span>
      {completed && <span className={styles.completed} aria-label="Tarefa completada" />}
    </li>
  )
}
