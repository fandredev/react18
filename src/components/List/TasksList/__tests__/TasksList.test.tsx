import { render, screen } from "@testing-library/react";
import TasksList from "..";


describe(`Test component: ${TasksList.name} `, () => {
  it('Should be possible to snapshot the TasksList component when the page is rendered', () => {
    const { container } = render(<TasksList id="1" selectTask={() => { }} time="01:00:00" task={'Aprender Vue'} completed={false} selected={false} />)

    expect(container).toMatchSnapshot()
  })

  it('Should only contain the "item" class when i don"t have the task completed or selected', () => {
    render(
      <TasksList
        id="1"
        selectTask={() => { }}
        time="01:00:00"
        task={'Aprender Vue'}
        completed={false}
        selected={false}
      />
    )

    const getClassList = screen.getByRole('listitem').getAttribute('class')
    expect(getClassList).toBe('item')

  })

  it('Should only contain the "item listSelected" classes when i have the task selected', () => {
    render(
      <TasksList
        id="1"
        selectTask={() => { }}
        time="01:00:00"
        task={'Aprender Vue'}
        completed={false}
        selected
      />
    )

    const getClassList = screen.getByRole('listitem').getAttribute('class')
    expect(getClassList).toMatch(/item listSelected/i)

  })

  it('Should only contain the "item listSelected listCompleted" classes when i have the task selected and completed', () => {
    render(
      <TasksList
        id="1"
        selectTask={() => { }}
        time="01:00:00"
        task={'Aprender Vue'}
        completed
        selected
      />
    )

    const getClassList = screen.getByRole('listitem').getAttribute('class')
    expect(getClassList).toMatch(/listSelected listCompleted/i)

  })

  it('The task that the user wants to do must be rendered', () => {
    render(
      <TasksList
        id="1"
        selectTask={() => { }}
        time="01:00:00"
        task={'Aprender Vue'}
        completed={false}
        selected={false}
      />
    )

    const getTitleItemList = screen.getByText('Aprender Vue')
    expect(getTitleItemList.textContent).toBe('Aprender Vue')
  })

  it('The time that the user wants to do must be rendered', () => {
    render(
      <TasksList
        id="1"
        selectTask={() => { }}
        time="01:00:00"
        task={'Aprender Vue'}
        completed={false}
        selected={false}
      />
    )

    const getTitleItemList = screen.getByText('01:00:00')
    expect(getTitleItemList.textContent).toBe('01:00:00')
  })

  it('Should show completed task icon when task is completed', () => {
    render(
      <TasksList
        id="1"
        selectTask={() => { }}
        time="01:00:00"
        task={'Aprender Vue'}
        completed
        selected={false}
      />
    )

    const getIconCompletedTask = screen.getByLabelText('Tarefa completada')
    expect(getIconCompletedTask).toBeInTheDocument()

  })

  it('Should be able to have the class "completed" when the task is completed', () => {
    render(
      <TasksList
        id="1"
        selectTask={() => { }}
        time="01:00:00"
        task={'Aprender Vue'}
        completed
        selected={false}
      />
    )

    const getIconCompletedTask = screen.getByRole('listitem').getAttribute('class')
    expect(getIconCompletedTask).toMatch(/listCompleted/i)

  })

})