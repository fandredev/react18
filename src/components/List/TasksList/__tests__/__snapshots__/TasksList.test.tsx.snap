// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component: TasksList  Should be possible to snapshot the TasksList component when the page is rendered 1`] = `
<div>
  <li
    class="item"
  >
    <h3>
      Aprender Vue
    </h3>
    <span>
      01:00:00
    </span>
  </li>
</div>
`;
